web/server/codechecker_server/api/report_server.py

```diff
@@ -3279,3 +3279,124 @@ def getAnalysisStatistics(self, run_id, run_history_id):
+                .filter(Report.bug_id.in_(comment_bug_ids)) \
+                .outerjoin(Report, Report.bug_id == Comment.bug_hash) \
+                .order_by(Comment.created_at.desc())
+
+            db_comments = {}
+            for comment in current_query:
+                db_comments[comment.id] = comment
+
+            for imported_comment in imported_comments:
+                # Check if the new comment exists in the database
+                if imported_comment.id in db_comments.keys():
+
+                    db_comment = db_comments.get(imported_comment.id)
+                    message = get_comment_msg(db_comment)
+
+                    # Check if the data is same for both
+                    if imported_comment.message != message:
+                        db_comment.message = imported_comment.message.encode(
+                            'utf-8')
+                        session.add(db_comment)
+
+                # If the comment does not exist add it
+                else:
+                    data = imported_data.get(imported_comment.id)
+                    new_comment = self.__add_comment(data[0], data[1].message)
+                    session.add(new_comment)
+
+            # Logic for importing review status
+            review_bug_ids = list(exportData.reviewData.keys())
+            review_query = session.query(Report.bug_id, ReviewStatus,
+                                         Report.id)\
+                .filter(Report.bug_id.in_(review_bug_ids)) \
+                .outerjoin(Report, Report.bug_id == ReviewStatus.bug_hash) \
+                .order_by(ReviewStatus.date.desc())
+
+            review_data = {}
+            for bug_hash, review, report_id in review_query:
+                review_data[bug_hash] = [review, report_id]
+
+            # print(review_data)
+
+            for bug_hash, imported_review in exportData.reviewData.items():
+
+                # If the review does not exist in the database
+                if bug_hash not in review_data:
+                    updated_status = imported_review.status
+                    updated_message = imported_review.comment
+
+                    report_id = session.query(Report.id)\
+                        .filter(Report.bug_id == bug_hash)\
+                        .all()
+
+                    report_id = report_id[0][0]
+
+                    res = self._setReviewStatus(
+                        report_id, updated_status, updated_message, session)
+
+                # Check if the date of the report is same
+                elif (str(review_data[bug_hash][0].date)
+                      != imported_review.date):
+                    updated_status = imported_review.status
+                    updated_message = imported_review.comment
+                    report_id = review_data[bug_hash][1]
+
+                    res = self._setReviewStatus(
+                        report_id, updated_status, updated_message, session)
```

Comment:
Here we get the report id because the `_setReviewStatus` requires it. First I recommend you to rebase your branch because there was an update in this function. Next we should change the parameter from report id to report hash: `def _setReviewStatus(self, session, report_hash, status, message, date=None):`. And there are already two places where we call this API function. The first place we have the `bug_id` what we can pass to this function so there is no need to get the report from the database. The other place we need to get the report from the database by report id and then pass the bug hash of this report to the set review status function:
https://github.com/Ericsson/codechecker/blob/b4f569dfa7b989424baf7876a96696cff38a7a55/web/server/codechecker_server/api/report_server.py#L1644

This way we can simplify this code on the following way:
```suggestion
                        review_bug_ids = list(exportData.reviewData.keys())
            review_query = session.query(ReviewStatus) \
                .filter(ReviewStatus.bug_hash.in_(review_bug_ids)) \
                .order_by(ReviewStatus.date.desc())

            db_review_data = {}
            for review_status in review_query:
                db_review_data[review_status.bug_hash] = review_status

            for bug_hash, imported_review in exportData.reviewData.items():
                db_status = db_review_data.get(bug_hash)

                # The status is up-to-date.
                if db_status and str(db_status.date) == imported_review.date:
                    continue

                date = datetime.strptime(imported_review.date,
                                         '%Y-%m-%d %H:%M:%S.%f')

                self._setReviewStatus(session,
                                      bug_hash,
                                      imported_review.status,
                                      imported_review.comment,
                                      date)
```